workbench-core/example/infrastructure/integration-tests/tests/datasets/create.test.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AddDataSetExternalEndpointResponse } from '@aws/workbench-core-datasets';
+import { v4 as uuidv4 } from 'uuid';
+import ClientSession from '../../support/clientSession';
+import Dataset from '../../support/resources/datasets/dataset';
+import Setup from '../../support/setup';
+import HttpError from '../../support/utils/HttpError';
+import RandomTextGenerator from '../../support/utils/randomTextGenerator';
+import { accessPointS3AliasRegExp, endpointIdRegExp } from '../../support/utils/regExpressions';
+
+describe('datasets create integration test', () => {
+  let setup: Setup;
+  let adminSession: ClientSession;
+  let randomTextGenerator: RandomTextGenerator;
+
+  let fakeDataSetId: string;
+
+  beforeEach(() => {
+    expect.hasAssertions();
+
+    fakeDataSetId = 'example-ds-badbadba-dbad-badb-adba-dbadbadbadba';
+  });
+
+  beforeAll(async () => {
+    setup = new Setup();
+    adminSession = await setup.getDefaultAdminSession();
+    randomTextGenerator = new RandomTextGenerator(adminSession.getSettings().get('runId'));
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('AddExternalEndpointForUser', () => {
+    let dataset: Dataset;
+    let userId: string;
+    let identityType: string;
+
+    beforeAll(async () => {
+      identityType = 'USER';
+      const { data } = await adminSession.resources.users.create({
```

Comment:
I have found this works better in the outer most `beforeAll()` block. That way, you can create one user and utilize it throughout the test suite.

Reply:
This is currently the only block that uses it. If another block does then it can be extracted.